tests/test_dummy.py

```diff
@@ -0,0 +1,543 @@
+    assert_anchor_group_equality(groups_of_3, [])
+
+    mol = Chem.MolFromSmiles("C(C)(F)CC")
+    core = [1, 0, 4, 3]
+    groups_of_1, groups_of_2, groups_of_3 = identify_anchor_groups(mol, core, 0)
+    assert_anchor_group_equality(groups_of_1, [[0]])
+    assert_anchor_group_equality(groups_of_2, [[0, 1], [0, 3]])
+    assert_anchor_group_equality(groups_of_3, [[0, 3, 4]])
+
+
+def test_enumerate_anchor_groups():
+    mol = Chem.MolFromSmiles("C1CC2OOC12")
+    core = [0, 1, 2, 5]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [3, 4])
+    assert_anchor_group_equality(groups_of_1, [[2], [5]])
+    assert_anchor_group_equality(groups_of_2, [[2, 1], [2, 5], [5, 0], [5, 2]])
+    assert_anchor_group_equality(groups_of_3, [[2, 1, 0], [2, 5, 0], [5, 0, 1], [5, 2, 1]])
+
+    # aspirin
+    mol = Chem.MolFromSmiles("CC(=O)OC1=CC=CC=C1C(=O)O")
+    core = [3, 4, 5, 6, 7, 8, 9, 10]
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2, 11, 12])
+    assert_anchor_group_equality(groups_of_1, [[3], [10]])
+    assert_anchor_group_equality(groups_of_2, [[3, 4], [10, 9]])
+    assert_anchor_group_equality(groups_of_3, [[3, 4, 5], [3, 4, 9], [10, 9, 8], [10, 9, 4]])
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [0, 1, 2])
+    assert_anchor_group_equality(groups_of_1, [[3]])
+    assert_anchor_group_equality(groups_of_2, [[3, 4]])
+    assert_anchor_group_equality(groups_of_3, [[3, 4, 5], [3, 4, 9]])
+    groups_of_1, groups_of_2, groups_of_3 = enumerate_anchor_groups(mol, core, [11, 12])
+    assert_anchor_group_equality(groups_of_1, [[10]])
+    assert_anchor_group_equality(groups_of_2, [[10, 9]])
+    assert_anchor_group_equality(groups_of_3, [[10, 9, 8], [10, 9, 4]])
+
+
+def test_enumerate_allowed_dummy_ixns_3_anchors():
+    dummy_group = {0, 1, 5, 8}
+    anchor_group = [2, 3, 4]
+
+    allowed_ixns = enumerate_dummy_ixns(dummy_group, anchor_group)
+
+    for ixn in allowed_ixns:
+        assert len(ixn) == len(set(ixn))
+
+    assert (0, 1, 2, 3) in allowed_ixns
+    assert (0, 1, 2) in allowed_ixns
+    assert (1, 2) in allowed_ixns
+    assert (0, 2, 3) in allowed_ixns
+    assert (1, 0, 2, 3) in allowed_ixns
+    assert (3, 2, 1, 5) in allowed_ixns
+    assert (1, 5, 2) in allowed_ixns
+    assert (2, 0, 5) in allowed_ixns
+    assert (1, 5, 8) in allowed_ixns
+    assert (1, 5, 8) in allowed_ixns
+    assert (0, 8, 5, 1) in allowed_ixns
+    assert (5, 1, 0, 8) in allowed_ixns
+    assert (3, 2, 8, 5) in allowed_ixns
+    assert (1, 8, 2, 3) in allowed_ixns
+
+    assert (0, 1, 3) not in allowed_ixns
+    assert (0, 1, 4) not in allowed_ixns
+    assert (0, 2, 4) not in allowed_ixns
+    assert (0, 3) not in allowed_ixns
+    assert (0, 4) not in allowed_ixns
+    assert (5, 3, 4) not in allowed_ixns
+    assert (1, 5, 4) not in allowed_ixns
+    assert (5, 8, 2, 4) not in allowed_ixns
+    assert (1, 8, 3, 4) not in allowed_ixns
```

Comment:
is it worth leaving a line comment for some of the trickier cases explaining why they're disallowed? A lot of these aren't obvious to me at this point

Reply:
I need to document these one by one... will get back to them